### PR TITLE
update the error type of blockchain's chain

### PR DESCRIPTION
**Pull request by NicolasDP**
remove one more dependency to error_chain crate

### CODE REVIEW

File: jormungandr/src/blockchain/chain.rs [hunk 1]
```diff
@@ -66,57 +66,97 @@ use tokio::stream::StreamExt;
...
-    }
+pub type Result<A> = std::result::Result<A, Error>;
 
-    errors {
-        Block0InitialLedgerError {
-            description("Error while creating the initial ledger out of the block0")
-        }
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("")]
```
Comment (mzabaluev) [inline jormungandr/src/blockchain/chain.rs:73]: You can use `#[error(transparent)]` on these

File: jormungandr/src/blockchain/chain.rs [hunk 2]
```diff
@@ -66,57 +66,97 @@ use tokio::stream::StreamExt;
...
+            message: message.to_owned(),
         }
     }
 }
 
+impl From<String> for Error {
+    fn from(message: String) -> Self {
+        Self::Custom { message }
+    }
+}
```
Comment (mzabaluev) [inline jormungandr/src/blockchain/chain.rs:158]: I think these two could be unified into `impl<T: Into<String>> From<T> for Error`, but this is finessing it.

Maybe even better to replace these with an `Error::custom` constructor to be able to audit uses, this looks like a stopgap when someone could not be bothered to create another error variant.

File: jormungandr/src/blockchain/chain.rs [hunk 3]
```diff
@@ -66,57 +66,97 @@ use tokio::stream::StreamExt;
...
 
-    errors {
-        Block0InitialLedgerError {
-            description("Error while creating the initial ledger out of the block0")
-        }
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("")]
+    Storage {
+        #[from]
```
Comment (eugene-babichenko) [inline jormungandr/src/blockchain/chain.rs:75]: `from` already implies `source`

**Comment by eugene-babichenko** [comment]
Superseded by #3310.

**Pull request closed without merging**